test/e2e_node/e2e_node_suite_test.go

```diff
@@ -117,10 +116,28 @@ var _ = BeforeSuite(func() {
 
 	// Reference common test to make the import valid.
 	commontest.CurrentSuite = commontest.NodeE2E
+
+	data, err := shared.Encode()
+	Expect(err).NotTo(HaveOccurred())
+
+	return data
+}, func(data []byte) {
+	// Set the shared context got from the synchronized initialize function
+	shared := &SharedContext{}
+	Expect(shared.Decode(data)).To(Succeed())
+	context = *shared
```

Comment:
This is used for parallel test suite.

For each parallel instance, the `BeforeSuite` will be run once.
However, we don't want to start kubelet, apiserver and etcd in each parallel instance. So we've to use `SynchronizedBeofreSuite`, in which the first function will only be called once, and the second function will be called in each parallel instance.